scripts/build-types.mjs

```diff
@@ -0,0 +1,54 @@
+import {cwd, isLaunchFile, write} from './script-utils.js'
+import { dirname, join, relative } from 'node:path'
+import { createRequire } from 'node:module'
+import { compileFromFile } from 'json-schema-to-typescript'
+
+const ROOT = join(cwd(import.meta.url), '..')
+const require = createRequire(import.meta.url);
+const configBuilderRoot = dirname(require.resolve("config-builder"));
+
+const defaultMapping = {
+    /**
+     * Add more mappings here
+     *  - `schema` should be an absolute path to a valid JSON Schema document
+     *  - `types` should be an absolute path to the output file
+     */
+    "Webcompat Settings": {
+        schema: join(configBuilderRoot, "tests/schemas/webcompat-settings.json"),
+        types: join(ROOT, "src/types/webcompat-settings.d.ts")
+    }
+}
+
+/**
+ * Read JSON schemas from the privacy-configuration repo and generate TypeScript types
+ *
+ * **note** we are NOT adding try/catch around each operation here since we want the script to fail fast
+ * and the errors given from node are sufficient to debug any issues
+ */
+export async function buildTypes(mapping = defaultMapping) {
+    for (let [featureName, manifest] of Object.entries(mapping)) {
+        const typescript = await compileFromFile(manifest.schema, {
+            bannerComment: `
+            // eslint-disable
+            /**
+             * @module ${featureName} Schema
+             * @description 
+             * These types are auto-generated from schema files within https://github.com/duckduckgo/privacy-configuration
```

Comment:
```suggestion
             * These types are auto-generated from schema files.
             * scripts/build-types.mjs is responsible for type generation.
             * See the privacy-configuration repo for the schema files:
             * https://github.com/duckduckgo/privacy-configuration
```

Reply:
Feel free to take this or leave it, it just makes the lines wrap a little less and is a little clearer where it's made from.

If you're taking it generate the type again :)